### PR TITLE
Updating web dependencies.

### PR DESCRIPTION
**Pull request by dennisoelkers**
- Updated webpack + loaders
- Updated typescript + loader
  - Removed static checkins of declaration files
  - Replaced them with `@types` packages
- Updated ESLint + configs from 3.x to 4.x

### CODE REVIEW

File: graylog2-web-interface/packages/eslint-config-graylog/package.json [hunk 1]
```diff
@@ -13,15 +13,15 @@
   "author": "Graylog, Inc. <hello@graylog.com>",
   "license": "MIT",
   "dependencies": {
-    "eslint": "^3.16.1",
-    "eslint-config-airbnb": "^14.1.0",
+    "eslint": "^4.3.0",
+    "eslint-config-airbnb": "^15.1.0",
```
Comment (edmundoa) [inline graylog2-web-interface/packages/eslint-config-graylog/package.json]: Do you know what changed there? I couldn't find any meaningful changelog. I only see more and more red in my editor, whenever I open a file, which is starting to be annoying for me :(

**Comment by edmundoa** [comment]
After disabling plugins I didn't notice anything odd, so I guess we just need to adapt the plugin configuration for these changes. In my opinion we should do PRs for that and merge everything once it's ready. Otherwise the build is going to be broken for a while and that's far from optimal.

**Comment by dennisoelkers** [comment]
@edmundoa: Did you try the graylog-project build with a clean setup (including cleaning all `node_modules` dirs)? As you can see from the [automated graylog-project build](https://jenkins.torch.sh/job/graylog-project-pr-snapshot/347/console) it should work for a clean, full build.

**Comment by dennisoelkers** [comment]
Updated the vendor module ids. Thanks! :+1:

**Pull request merged**